include/RevHart.h

```diff
@@ -26,7 +26,7 @@ class RevHart{
   EcallState Ecall{};
 
   ///< RevHart: Pointer to the Proc's LSQueue
-  const std::shared_ptr<std::unordered_map<uint64_t, MemReq>>& LSQueue;
+  const std::shared_ptr<std::multimap<uint64_t, MemReq>>& LSQueue;
```

Comment:
There's an `std::unordered_multimap` if you want to retain `O(1)` amortized lookups. I would do a performance comparison between `std:multimap` and `std::unordered_multimap`.

Reply:
equivalent functionality between unordered_multimap and multimap for this application so if we get better performance that works. updated